Sources/Controllers/MessagesViewController.swift

```diff
@@ -82,25 +81,24 @@ UICollectionViewDelegateFlowLayout, UICollectionViewDataSource {
         setupDelegates()
         addMenuControllerObservers()
         addObservers()
+        addKeyboardObservers()
     }
-    
+
     open override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         isMessagesControllerBeingDismissed = false
+        messageCollectionViewBottomInset = keyboardOffsetFrame.height - safeAreaBottomInset
+        messagesCollectionView.scrollToBottom(animated: true)
     }
-    
+
     open override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         isMessagesControllerBeingDismissed = true
+        messageCollectionViewBottomInset = keyboardOffsetFrame.height - safeAreaBottomInset
+        messagesCollectionView.scrollToBottom(animated: true)
     }
-    
+
     open override func viewDidLayoutSubviews() {
-        // Hack to prevent animation of the contentInset after viewDidAppear
-        if isFirstLayout {
-            defer { isFirstLayout = false }
-            addKeyboardObservers()
```

Comment:
Since we unobserve in `deinit`, we should pair and start observation in init or viewDidLoad - not sure why we would want to do it here, so corrected

Reply:
I can’t really see the diff well for this in GitHawk, but certain keyboard events we only observe once to get the behavior we want